explorer/interpreter/type_checker.cpp

```diff
@@ -3978,10 +3981,9 @@ auto TypeChecker::TypeCheckPattern(
           << "conversion to type succeeded but didn't produce a type, got "
           << *type;
       if (expected) {
-        if (IsConcreteType(type)) {
-          CARBON_RETURN_IF_ERROR(ExpectType(p->source_loc(), "name binding",
-                                            type, *expected, impl_scope));
-        } else {
+        // TODO: Per #2188, we should be performing conversions at this level
```

Comment:
Minor, but I wonder if it'd help to say "proposal #2188" to make it clearer that your mentions aren't referring to an issue, for anyone who may not have the same immersion that we do. :)